### DISCUSSION TITLE
Compile problem any EthernetWebServer example in VS CODE platformIO

### DISCUSSION THREAD

**Discussion by dirkdickler** [khoih-prog/EthernetWebServer#57] [General]
Hello,  I have problem with succes compile any project example - during build I get lot of errors :-(
I spend lot of days with this. but I'm really KO :-(.
My case is ESP32 + W5500 
My please is anybody have any succces compile project from this examples ?  Can you send me it? packing to ZIP file a send me example folder, pleaseeee  help mee :-) 
Thank you for help and my email is : rado.semansky@gmail.com 

![ESP32 compile problem](https://user-images.githubusercontent.com/6540400/167310406-ed4f86f0-0ce3-4bd6-9dcc-b7b87eb6ed86.JPG)

**Comment by khoih-prog** [comment]
It's possible you didn't use the `Libraries' Patches` to fix the ESP32 core issue

Check [**For fixing ESP32 compile error**](https://github.com/khoih-prog/EthernetWebServer#7-for-fixing-esp32-compile-error)

---

#### 7. For fixing ESP32 compile error

To fix [`ESP32 compile error`](https://github.com/espressif/arduino-esp32), just copy the following file into the [`ESP32`](https://github.com/espressif/arduino-esp32) cores/esp32 directory (e.g. ./arduino-1.8.19/hardware/espressif/cores/esp32) to overwrite the old file:
- [Server.h](LibraryPatches/esp32/cores/esp32/Server.h)
---

Compile OK with Adafruit Feather ESP32 V2

![Selection_001](https://user-images.githubusercontent.com/57012152/167319697-5aa48d5d-8c93-4c88-802e-0b6074147c76.png)

**Comment by dirkdickler** [comment]
Hello, problem was   Server.h  file .   Important is modify complet server.h file as is write here in documentation. :-)

**Comment by th-hock** [comment]
Hi,
I'm also facing a compile issue. All the examples I tried (e.g. AdvancedWebServer.ino), resulted in the following error:
~~~~~~~~~~~~~~~
C:\Users\xxx\Documents\Arduino\libraries\EthernetWebServer\src/EthernetWebServer.hpp:312:12: fatal error: string: No such file or directory
   #include <string>
            ^~~~~~~~
compilation terminated.
exit status 1
~~~~~~~~~~~~~~~

I'm using the Arduino IDE version 1.8.19 (AVR Boardmanager version 1.8.6).
Target platform is an Arduino AVR Board (Arduino Nano) and a enc28j60 ethernet shield.
I'm using the EthernetENC library and enabled "#define USE_ETHERNET_ENC true" accordingly.
If I got your great documentation right, no Packages_Patches and no LibaryPatches are required for this combination.

Maybe it is just a stupid configuration issue, but I'm not able to solve it myself. Sorry.
Any hint is appreciated. Thanks a lot in advance.

**Comment by khoih-prog** [comment]
> Target platform is an Arduino AVR Board (Arduino Nano) and a enc28j60 ethernet shield.

**UNO, Nano** are not supported. Only Mega and better board. As a normal practice, always check [**Currently supported Boards**](https://github.com/khoih-prog/EthernetWebServer#currently-supported-boards).

**Reply by th-hock** [reply]
ok, thanks. I checked the board list, but since there are multiple Nano variants listed I didn't realize that the plain one is no supported.
Sorry.
